configure.ac

```
@@ -73,6 +73,17 @@ AC_DEFUN([PY_CHECK_FUNC],
   AS_VAR_POPDEF([py_define])
 ])
 
+dnl PY_CHECK_CPP(CPP_CHECK, [LIST-OF-INCLUDES], [ACTION-IF-FOUND], [ACTION-IF-NOT-FOUND])
+AC_DEFUN([PY_CHECK_CPP],
+         [AC_PREPROC_IFELSE([AC_LANG_PROGRAM([
+m4_foreach_w([incl], [$2], [@%:@include <incl> m4_newline])
+@%:@if $1
+/* pass */
+#else
+#error "not defined"
```

This function is more general than just checking for a definition.
```suggestion
dnl PY_CHECK_CPP(CPP_CHECK, [LIST-OF-INCLUDES], [ACTION-IF-TRUE], [ACTION-IF-FALSE])
AC_DEFUN([PY_CHECK_CPP],
         [AC_PREPROC_IFELSE([AC_LANG_PROGRAM([
m4_foreach_w([incl], [$2], [@%:@include <incl> m4_newline])
@%:@if $1
/* pass */
#else
#error "check returned false"
```

---

Thanks, good observation!